src/Persistence/Reflection/RuntimeReflectionProperty.php

```diff
@@ -56,7 +56,7 @@ public function getValue($object = null)
     #[ReturnTypeWillChange]
     public function setValue($object, $value = null)
     {
-        if (! ($object instanceof Proxy && ! $object->__isInitialized())) {
+        if ((! ($object instanceof Proxy && ! $object->__isInitialized())) || $this->isStatic()) {
```

Comment:
The extra parenthesis are not necessary. You can switch the operands to remove any ambiguity.
```suggestion
        if ($this->isStatic() || ! ($object instanceof Proxy && ! $object->__isInitialized())) {
```